### PR TITLE
fix to pipeline

### PR DESCRIPTION
**Pull request by sithmel**
see #137 
Type declaration is missing

### CODE REVIEW

File: src/pipeline.mjs [hunk 1]
```diff
@@ -1,3 +1,3 @@
-export default function pipeline (...fns) {
-  return fns.reverse().reduce((f, g) => (...args) => f(g(...args)))
+export default function pipeline (iterable, ...fns) {
```
Comment (KSXGitHub) [inline src/pipeline.mjs]: About the argument name "iterable": While users of `iter-tools` mostly focus on iterables, this function is applicable to other types as well, the following code is valid for example:

```javascript
const result = pipeline(
  3,
  x => x * 2,
  x => x + 1,
  x => x * x
)
```

**Comment by KSXGitHub** [comment]
Should we rename the function to `pipe`? It is shorter, it is also more familiar as other FP libraries provide `pipe`.

File: src/__test__/pipeline.test.js [hunk 2]
```diff
@@ -11,12 +11,7 @@ function isEven (x) {
 
 describe('pipeline', function () {
   it('using function', function () {
```
Comment (KSXGitHub) [inline src/__test__/pipeline.test.js]: This line is not only no longer necessary, it is somewhat misleading

**Comment by sithmel** [comment]
@KSXGitHub please fix the type declaration. I'll take care of the rest and to merge

**Comment by KSXGitHub** [comment]
> @KSXGitHub please fix the type declaration. I'll take care of the rest and to merge

https://github.com/sithmel/iter-tools/pull/142

(How did you miss that?)

**Comment by KSXGitHub** [comment]
BTW, https://github.com/sithmel/iter-tools/pull/142 assumes that you use the name `pipeline` just in case you want to rename the function to `pipe`.

**Pull request merged**